apps/system/js/places.js

```diff
@@ -295,20 +379,71 @@
      */
     isPinned: function(url) {
       return new Promise((resolve, reject) => {
-        return this.getStore()
-          .then(store => {
-            return store.get(url);
-          })
-          .then(place => {
+        return this.getDb().then(db => {
+          var transaction = db.transaction(this.PAGES_STORE, 'readonly');
+          var objectStore = transaction.objectStore(this.PAGES_STORE);
+          var request = objectStore.get(url);
+          request.onsuccess = function() {
+            var place = request.result;
             return resolve(!!place.pinned);
-          })
-          .catch(e => {
+          };
+          request.onerror = function(e) {
             console.error(`Error getting the page details: ${e}`);
             return reject(e);
-          });
+          };
+        });
       });
     },
 
+    /**
+     * Pin a site.
+     *
+     * @param {String} id Site ID.
+     * @param {Object} siteObject Site object.
+     */
+    pinSite: function(id, siteObject) {
+      return new Promise((function(resolve, reject) {
+        this.getDb().then((function(db) {
+          var transaction = db.transaction(this.SITES_STORE, 'readwrite');
+          var objectStore = transaction.objectStore(this.SITES_STORE);
+          var writeRequest = objectStore.put(siteObject);
+        
+          writeRequest.onsuccess = function() {
+            console.log('Successfully pinned site ' + ' with id ' +
+              siteObject.id);
+            resolve();
+          };
+    
+          writeRequest.onerror = function() {
+            console.error('Error updating site with id ' + siteObject.id);
+            reject();
+          };
+        }).bind(this));
+      }).bind(this));
+    },
+
+    /**
+     * Get all pinned sites.
+     *.
+     * @returns {Promise} A promise which resolves with the full set of results.
+     */
+    getPinnedSites: function() {
+      var results = [];
+      var transaction = this.db.transaction(this.SITES_STORE);
+      var objectStore = transaction.objectStore(this.SITES_STORE);
+      return new Promise((function(resolve, reject) {
+        objectStore.openCursor().onsuccess = function(event) {
```

Comment:
Is not this returning all the sites, and not only the pinned ones? If the idea is that all the sites stored in the SITES_STORE are pinned, can we remove the index?